sharding/expandedListSampling_test.go

```diff
@@ -0,0 +1,150 @@
+	weights[1] = 0
+	hasher := &sha256.Sha256{}
+
+	selector, err := NewSelectorExpandedList(validators, weights, hasher)
+	require.Equal(t, ErrInvalidWeight, err)
+	require.Nil(t, selector)
+}
+
+func TestNewSelectorExpandedListOK(t *testing.T) {
+	t.Parallel()
+
+	validators := createDummyNodesList(10, "shard_0")
+	weights := createDummyWeights(10, 32, 2, nextWeightArithmeticProgression)
+	hasher := &sha256.Sha256{}
+
+	selector, err := NewSelectorExpandedList(validators, weights, hasher)
+	require.Nil(t, err)
+	require.NotNil(t, selector)
+
+	for i := uint32(0); i < uint32(len(validators)); i++ {
+		offset := uint32(0)
+		if i > 0 {
+			offset = weights[i-1]
+		}
+		for j := i + offset; j < weights[i]; j++ {
+			require.Equal(t, selector.expandedList[j], i)
+		}
+	}
+}
+
+func TestSelectorExpandedList_IsInterfaceNil(t *testing.T) {
+	t.Parallel()
+
+	var selector RandomSelector
+	require.True(t, check.IfNil(selector))
+
+	validators := createDummyNodesList(10, "shard_0")
+	weights := createDummyWeights(10, 32, 2, nextWeightGeometricProgression)
+	hasher := &sha256.Sha256{}
+
+	selector, _ = NewSelectorExpandedList(validators, weights, hasher)
+	require.False(t, check.IfNil(selector))
+}
+
+func TestSelectorExpandedList_SelectNilRandomessShouldErr(t *testing.T) {
+	t.Parallel()
+
+	validators := createDummyNodesList(10, "shard_0")
+	weights := createDummyWeights(10, 32, 2, nextWeightGeometricProgression)
+	hasher := &sha256.Sha256{}
+
+	selector, _ := NewSelectorExpandedList(validators, weights, hasher)
+	indexes, err := selector.Select(nil, 5)
+
+	require.Equal(t, ErrNilRandomness, err)
+	require.Nil(t, indexes)
+}
+
+func TestSelectorExpandedList_Select0SampleSizeShouldErr(t *testing.T) {
+	t.Parallel()
+
+	validators := createDummyNodesList(10, "shard_0")
+	weights := createDummyWeights(10, 32, 2, nextWeightGeometricProgression)
+	hasher := &sha256.Sha256{}
+
+	selector, _ := NewSelectorExpandedList(validators, weights, hasher)
+	indexes, err := selector.Select([]byte("random"), 0)
+
+	require.Equal(t, ErrInvalidSampleSize, err)
+	require.Nil(t, indexes)
+}
+
+func TestSelectorExpandedList_SelectSampleSizeGreaterThanSetShouldErr(t *testing.T) {
+	t.Parallel()
+
+	setSize := uint32(10)
+	validators := createDummyNodesList(setSize, "shard_0")
+	weights := createDummyWeights(setSize, 32, 2, nextWeightGeometricProgression)
+	hasher := &sha256.Sha256{}
+
+	selector, _ := NewSelectorExpandedList(validators, weights, hasher)
+	indexes, err := selector.Select([]byte("random"), setSize+1)
+
+	require.Equal(t, ErrInvalidSampleSize, err)
+	require.Nil(t, indexes)
+}
+
+func TestSelectorExpandedList_SelectOK(t *testing.T) {
+	t.Parallel()
+
+	setSize := uint32(10)
+	validators := createDummyNodesList(setSize, "shard_0")
+	weights := createDummyWeights(setSize, 32, 2, nextWeightGeometricProgression)
+	hasher := &sha256.Sha256{}
+	sampleSize := setSize / 2
+
+	selector, _ := NewSelectorExpandedList(validators, weights, hasher)
```

Comment:
Can you pls add a test where the validators and weights have different lengths: one time len(validators) > len(weights) and one time len(validators) < len(weights)

Reply:
removed the validators list, no longer required.

Reply:
ok